### PR TITLE
switch defaults from generating unstable Rust to generating stable Rust

### LINKED ISSUE #757: Switch defaults from generating unstable Rust to generating stable Rust
The current default is definitely the wrong choice.

This involves 

* changing the `Builder::no_unstable_rust` method to the `Builder::unstable_rust` method

* changing the `--no-unstable-rust` flag to a `--unstable-rust` flag in `src/options.rs`

* I don't think this will require updating test results since we run all the tests with `--no-unstable-rust`, however it may require removing that flag from the test harness. I sugget `git grep`ing for "unstable" to see places in the code base that may require updating.

### PR DESCRIPTION
**Pull request by rigelk**
As said in the issue:
- changing the Builder::no_unstable_rust method to the Builder::unstable_rust method
- changing the --no-unstable-rust flag to a --unstable-rust flag in src/options.rs
- changing bindgen-flags header in the test headers to remove the --no-unstable-rust flag
- removing --no-unstable-rust flag in ./test/test-one.sh

Fixes #757

r? @fitzgen

### CODE REVIEW

File: ir.dot [hunk 1]
```diff
@@ -0,0 +1,120 @@
+digraph {
```
Comment (emilio) [inline ir.dot]: These should go away before landing.

File: src/lib.rs [hunk 2]
```diff
@@ -736,8 +736,8 @@ impl Builder {
     }
 
     /// Avoid generating any unstable Rust, such as Rust unions, in the generated bindings.
-    pub fn no_unstable_rust(mut self) -> Builder {
-        self.options.unstable_rust = false;
+    pub fn unstable_rust(mut self) -> Builder {
```
Comment (emilio) [inline src/lib.rs]: While we're breaking the API, should we take a boolean just for consistency?

i.e., `fn unstable_rust(mut self, doit: bool) -> Self`?

Comment (fitzgen) [inline src/lib.rs]: +1 this is what we do elsewhere, and consistency is great

File: ir.dot (hunk 1 above)
Comment (rigelk) [inline ir.dot]: oh! sorry, that's me not reseting the files after the test compilation!

File: src/lib.rs (hunk 2 above)
Comment (rigelk) [inline src/lib.rs]: looks way cleaner indeed! I modified the commit accordingly

File: book/src/using-unions.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Using the Union Types Generated by Bindgen
 
-**NOTE:** As of Rust version 1.17, Rust does not have a stable `union` type. Issue [#32836](https://github.com/rust-lang/rust/issues/32836) tracks the stabilization of a `union` type in Rust. By default, bindgen will generate the preliminary unstable `union` type, unless the flag `--no-unstable-rust` flag is used.
+**NOTE:** As of Rust version 1.17, Rust does not have a stable `union` type. Issue [#32836](https://github.com/rust-lang/rust/issues/32836) tracks the stabilization of a `union` type in Rust. 
+
+By using the flag `--unstable-rust`, bindgen will generate the preliminary unstable `union` type.
```
Comment (fitzgen) [inline book/src/using-unions.md]: Thanks for updating the book :)

**Pull request merged**